controllers/upload.go

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"zestream-server/constants"
+	"zestream-server/types"
+	"zestream-server/utils"
+)
+
+func GeneratePresignedURL(c *gin.Context) {
+
+	// Validate the incoming request
+	var request types.GenerateURLRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Incorrect request parameters", "details": err.Error()})
+		return
+	}
+
+	// Create a new session
+	sess, _ := session.NewSession(&aws.Config{
+		Region: aws.String(constants.S3_REGION),
+	})
+	svc := s3.New(sess)
+
+	// Generate a New Video ID
+	videoID := utils.VideoIDGen(request.FileName)
```

Comment:
can we extract extension from fileName then add it to videoID

Reply:
As it is a GET request we may not receive any file extension in the body for some cases like the one in above screen-shot so to counter that i've avoided the use of file extension